code/game/objects/structures/watercloset.dm

```diff
@@ -486,8 +486,22 @@
 		new_sink.setDir(dir)
 		qdel(src)
 		return
+	if(I.tool_behaviour == TOOL_WRENCH && !(flags_1 & NODECONSTRUCT_1))
+		I.play_tool_sound(src)
+		deconstruct()
+		return
+
 	return ..()
 
+/obj/structure/sinkframe/deconstruct()
+	if(!(flags_1 & NODECONSTRUCT_1))
+		drop_materials()
+	..()
+
+/obj/structure/sinkframe/proc/drop_materials()
+	for(var/datum/material/mat in custom_materials)
```

Comment:
```suggestion
	for(var/datum/material/mat as anything in custom_materials)
```
Unless the list has multiple types and you're looking for a specific one, avoid type-checking.